### PR TITLE
Moving Kafka apps to a dedicated node

### PR DESCRIPTION
**Pull request by sbuliarca**

### CODE REVIEW

File: helm/kafka/templates/kafka-proxy-deployment.yaml [hunk 1]
```diff
@@ -19,6 +19,22 @@ spec:
         app: {{ .Values.service.name }}-rest-proxy
         visualize: "true"
     spec:
+      tolerations:
+      # Making sure that Kafka-proxy will be scheduled on dedicated nodes for kafka. These nodes are tainted by the provisioner.
+      - key: "dedicated"
+        operator: "Equal"
+        value: "kafka"
+        effect: "NoSchedule"
+
```
Comment (tamas-molnar) [inline helm/kafka/templates/kafka-proxy-deployment.yaml]: please remove blank line

File: helm/kafka/templates/kafka-statefulset.yaml [hunk 2]
```diff
@@ -19,6 +19,22 @@ spec:
...
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: kube-aws.coreos.com/role
+                operator: In
+                values:
+                - kafka-worker
+
```
Comment (tamas-molnar) [inline helm/kafka/templates/kafka-statefulset.yaml]: please remove blank line

File: helm/kafka/templates/zookeeper-deployment.yaml [hunk 3]
```diff
@@ -19,6 +19,22 @@ spec:
...
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: kube-aws.coreos.com/role
+                operator: In
+                values:
+                - kafka-worker
+
```
Comment (tamas-molnar) [inline helm/kafka/templates/zookeeper-deployment.yaml]: please remove blank line

File: helm/kafka/templates/kafka-proxy-deployment.yaml [hunk 4]
```diff
@@ -19,6 +19,22 @@ spec:
         app: {{ .Values.service.name }}-rest-proxy
         visualize: "true"
     spec:
+      tolerations:
```
Comment (tamas-molnar) [inline helm/kafka/templates/kafka-proxy-deployment.yaml]: where can this be seen on the nodes?

Comment (sbuliarca) [reply helm/kafka/templates/kafka-proxy-deployment.yaml]: This shows up at a kubectl describe nodes. I had to upgrade kubectl to latest to show up

**Pull request merged**